### PR TITLE
Persist new articles

### LINKED ISSUE #12: Add ability to add a article
> As a general user, I want the ability to add a topic , so that I can discuss legislation and other organizational ideas.

### Acceptance Criteria:
- [x] Can create a new article when signed in
    - [x] Title field
    - [x] Content text field 
- [ ] Cannot **in any way** create an article when not signed in
- [ ] Cannot impersonate writing an article as another user
- [ ] Validates the article has a title
- [ ] Displays errors to the user

#### Not in scope
- Handling multiple requests for the same article creation (idempotency)
- Other article fields

### Tasks:
- [x] Function to persist to database
- [x] Api layer to transform request to data for persistance
- [ ] Return appropriate errors
- [x] Update UI to store article form inputs state (should this be in the global atom?)
- [x] Call the api with the correct data

### PR DESCRIPTION
**Pull request by wdoug**
Closes #12

### CODE REVIEW

**Comment by dviramontes** [comment]
@wdoug i rebased this from origin/master, just FYI

File: src/rmfu/core.clj [hunk 1]
```diff
@@ -133,6 +132,19 @@
                                 (if (and unsigned-token (:is-admin? unsigned-token) (db/block-user email state))
                                   (ok (if state "blocked" "unblocked"))
                                   (unauthorized {:error "not authorized to block users"}))))
+                        auth-backend)
+
+                      (wrap-authentication
```
Comment (wdoug) [inline src/rmfu/core.clj]: Should we move this up above the `;; ADMIN ONLY ROUTES` comment?

Comment (dviramontes) [inline src/rmfu/core.clj]: totally, got it locally, sorry
i missed it.

File: src/rmfu/persistance.clj [hunk 2]
```diff
@@ -113,4 +113,17 @@
...
\ No newline at end of file
+      (mc/update-by-id db coll oid {$set {:blocked? state}}))))
+
+(defn add-article!
+  "Adds a new article to the database.
+  Warning: This method is not idempotent currently."
+  [article author_id]
+  (let [{:keys [title content]} article
+        article-oid (ObjectId.)
+        article-doc {:author_id author_id
```
Comment (wdoug) [inline src/rmfu/persistance.clj]: @dviramontes how do you feel about calling this `:author-email` instead of `:author_id`? For me that would imply the `:_id` of the author user doc.

Comment (dviramontes) [inline src/rmfu/persistance.clj]: :+1:  changing it now

File: src/rmfu/persistance.clj [hunk 3]
```diff
@@ -113,4 +113,17 @@
         coll "users"
```
Comment (hasuf) [inline src/rmfu/persistance.clj]: Comment on the overall file: We should spell persistence correctly. :)

Comment (wdoug) [inline src/rmfu/persistance.clj]: Ah whoops. We can do that in another pull request.

**Comment by dviramontes** [comment]
:+1:
